File: backend/blackbox/src/main/java/com/insight_web/blackbox/usecase/storage/SiteStorage.java
package com.insight_web.blackbox.usecase.storage;

import com.insight_web.blackbox.domain.Site;

import java.util.List;

public interface SiteStorage {
    Site findById(String siteId);
    List<Site> findAllByOwnerId(Long ownerId);
    Site create(Site site);
}

File: backend/auth/src/main/kotlin/com/insightweb/auth/usecase/storage/SiteStorage.kt
package com.insightweb.auth.usecase.storage

import com.insightweb.auth.domain.Site

interface SiteStorage {
    fun findById(siteId: String): Site
    fun findAllByOwnerId(ownerId: Long): List<Site>
    fun create(site: Site): Site
}